addons/event_sms/models/event_registration.py

```diff
@@ -1,13 +1,62 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import models
+from odoo import api, models
+from odoo.addons.phone_validation.tools import phone_validation
 
 
 class Registration(models.Model):
     _inherit = 'event.registration'
 
+    @api.onchange('phone', 'event_id', 'partner_id')
+    def _onchange_phone_validation(self):
+        if self.phone:
+            country = self.partner_id.country_id or self.eventid.country_id or self.env.company.country_id
+            self.phone = self._phone_format(self.phone, country)
+
+    @api.onchange('mobile', 'event_id', 'partner_id')
+    def _onchange_mobile_validation(self):
+        if self.mobile:
+            country = self.partner_id.country_id or self.eventid.country_id or self.env.company.country_id
+            self.mobile = self._phone_format(self.mobile, country)
+
+    @api.model_create_multi
+    def create(self, values_list):
+        """ Override to try to format mobile or phone numbers, as it is used
+        notably to send reminders. Having international numbers eases process
+        of preparing and sending SMS. """
+        for values in values_list:
+            if not values.get('phone') and not values.get('mobile'):
+                continue
+
+            related_country = self.env['res.country']
+            if values.get('partner_id'):
+                related_country = self.env['res.partner'].browse(values['partner_id']).country_id
```

Comment:
How about accumulating the partners / events to browse and fetch them all at once to avoid a lot of small queries?
(You could accumulate and fetch only for the ones that need to have their phones/mobiles formatted).